src/analysis/type_analysis.cpp

```diff
@@ -508,12 +508,20 @@ class BasicBlockTypePropagator : public ExprVisitor, public StmtVisitor, public
 
     void* visit_unaryop(AST_UnaryOp* node) override {
         CompilerType* operand = getType(node->operand);
+        if (!hasFixedOps(operand))
+            return UNKNOWN;
 
         // TODO this isn't the exact behavior
         BoxedString* name = getOpName(node->op_type);
         CompilerType* attr_type = operand->getattrType(name, true);
+
+        if (attr_type == UNDEF)
+            attr_type = UNKNOWN;
```

Comment:
This feels odd to me -- wouldn't it be getattrType's fault more than this function's?